pkg/ucp/integrationtests/resourceproviders/resourceproviders_test.go

```diff
@@ -87,3 +94,31 @@ func Test_ResourceProvider_CascadingDelete(t *testing.T) {
 	response.EqualsErrorCode(404, "NotFound")
 	require.Equal(t, locationID, response.Error.Error.Target)
 }
+
+func Test_ResourceProvider_RegisterManifests(t *testing.T) {
+	server := testhost.Start(t, testhost.TestHostOptionFunc(func(options *ucp.Options) {
+		options.Config.Initialization.ManifestDirectory = "testdata/manifests"
+	}))
+
+	createRadiusPlane(server)
+
+	// Wait for the manifest to be registered
+	time.Sleep(5 * time.Second)
```

Comment:
Can we do this with `require.Eventually` instead? 

Picking a hardcoded value has a lot of downsides:

- If it's too short, the test will be unreliable
- If it's too long, we end up waiting a lot

You can use `require.Eventually` with a long-ish timeout and it will be mitigate both of these issues. eg: retry every 1 second with a 30 second deadline.